.github/workflows/build-workspace.yml

```diff
@@ -17,6 +17,7 @@ jobs:
     concurrency:
       group: build-workspace-analyze-${{ inputs.ref || github.event.pull_request.number || github.ref }}
       cancel-in-progress: true
+    continue-on-error: true
```

Comment:
This means the job would not fail if a step fails. See: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

I think we should not enable this flag as it might result in a passed check when there are errors.

Reply:
I see, I didn't consider how the failures would propagate to the job, thank you for looking for that 🙌 

What about using `if: ${{ !cancelled() }}` from the second to the last step in that job then?  
From what I've read about it, it should address our case. Initially I wanted to got for [`if: always()`](https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f), but the official docs suggest this [`!cancelled()`](https://docs.github.com/en/actions/learn-github-actions/expressions#always) expression

The proposed changes are addressed here: b9ca9907336fb76cdde229e24981e2c5103cc1d0

Reply:
I'm still not sure, that means if a later step passes, the job is green.
I would keep it as it is for now.
Local linting in pre-commit hook should catch the problems first anyway.

Reply:
Yes, that's true

Reverted the changes in f79aa1c6b88c39ab28efdd2585d96b8d14505460